pkg/kubectl/cluster.go

```diff
@@ -30,6 +31,8 @@ import (
 type ClusterGeneratorV1Beta1 struct {
 	// Name of the cluster context (required)
 	Name string
+	// Name of the federation this cluster is joining (required)
```

Comment:
kubefed should not be using ClusterGenerator? That seems like a separate issue. Is it being tracked somewhere?

I'll revert all of the code that adds the annotation to the cluster objects later today unless someone objects. We can always add it back later.